option.go

```diff
@@ -119,6 +121,24 @@ func EqualFuncs(b bool) Option {
 	}}
 }
 
+// KeepFields transforms a value of struct type T. It makes a copy of its input
+// and sets non-specified fields to their zero values. See also Transform.
+func KeepFields[T any](fields ...string) Option {
```

Comment:
IMO this should check that the given fields are actually in the struct, just like ZeroFields does. This one is even more important because if you have a typo in this one, it won't even panic during comparison, it'll just do the wrong thing.